app/src/ui/stashing/stash-diff-viewer.tsx

```diff
@@ -94,20 +94,26 @@ const Header: React.SFC<{
   repository: Repository
   dispatcher: Dispatcher
 }> = props => {
-  const onClearClick = () => {
+  const onDiscardClick = () => {
     props.dispatcher.dropStash(props.repository, props.stashEntry)
   }
-  const onSubmitClick = () => {
+  const onRestoreClick = () => {
     props.dispatcher.popStash(props.repository, props.stashEntry)
   }
   return (
     <div className="header">
       <h3>Stashed changes</h3>
-      <ButtonGroup destructive={true}>
-        <Button onClick={onSubmitClick} type="submit">
+      {/*
+      this is false because ButtonGroup assumes the submit button
+      performs the destructive action. In this case the destructive
+      action is performed by the non-submit button so we lie to the
+      props to get the correct button ordering
+       */}
+      <ButtonGroup destructive={false}>
```

Comment:
I think this may work just fine without the destructive false, unless I'm missing something?

Reply:
it would! i wanted to pass it explicitly (and with the comment above it) so the intent was super clear 😄

Reply:
Moving this comment outside of the return statement will help make the code a bit more readable.

Reply:
done in https://github.com/desktop/desktop/pull/7392/commits/bb022a458aac3d3a75188306c4ec12fc574b0cbc